service_test.go

```diff
@@ -332,6 +336,81 @@ var _ = Describe("Service", func() {
 		})
 	})
 
+	// inspired from https://golang.org/src/net/http/serve_test.go?#L505
+	Describe("Server", func() {
+		fmt.Println(GinkgoWriter, "Testing Server")
+		const appName = "foo"
+		var s *goa.Service
+		var reqNum int
+		var handler goa.Handler
+		var timeout time.Duration
+
+		BeforeEach(func() {
+			s = goa.New(appName)
+			timeout = 500 * time.Millisecond
+			s.Server.ReadTimeout = timeout
+
+			handler = func(c context.Context, rw http.ResponseWriter, req *http.Request) error {
+				reqNum++
+				rw.WriteHeader(200)
+				rw.Write([]byte(fmt.Sprintf("req=%d", reqNum)))
+				return nil
+			}
+
+			ctrl := s.NewController("test")
+			s.Mux.Handle("GET", "/", ctrl.MuxHandler("get", handler, nil))
+
+			// note: should we be using an httptest / ghttp server, here?
```

Comment:
Yes ghttp would indeed work well here - it would get rid of the race condition. https://godoc.org/github.com/onsi/gomega/ghttp has an example on how to use it also the [docs](https://onsi.github.io/gomega/#ghttp-testing-http-clients) are pretty good.

Reply:
Hmm.. I'll have to think about this more. The problem is that `s.Server` is an `*http.Server`, not `*httptest.Server` nor `*ghttp.Server` so I can't assign those to it.